java/compaction/compaction-job-execution/src/test/java/sleeper/compaction/job/execution/CompactionTaskTest.java

```diff
@@ -113,8 +125,9 @@ void shouldTerminateIfNoJobsArePresentAfterRunningForIdleTime() throws Exception
             // Given
             instanceProperties.setNumber(COMPACTION_TASK_MAX_IDLE_TIME_IN_SECONDS, 3);
             Queue<Instant> times = new LinkedList<>(List.of(
-                    Instant.parse("2024-02-22T13:50:00Z"),
-                    Instant.parse("2024-02-22T13:50:03Z")));
+                    Instant.parse("2024-02-22T13:50:00Z"), // Start
+                    Instant.parse("2024-02-22T13:50:03Z"),
+                    Instant.parse("2024-02-22T13:50:07Z"))); // Finish
```

Comment:
It's difficult to be confident here about which of these times are actually used for the idle time check. This problem would go away if we avoid checking the time again after the idle time check fails. There are similar problems in other tests.